virtualbox.py

```diff
@@ -0,0 +1,78 @@
+import subprocess
+from pathlib import Path
+from sys import platform
+import datetime
+from pytz import UTC
+
+utc = UTC
+
+class virtualbox(object):
+
+    def __init__(self, virtualboxpath, tz) -> None:
+        if platform.lower() == "win32":
+            virtualboxpath = virtualboxpath.replace('\"', "")
+            virtualboxpath = virtualboxpath.replace("\'", "")
+            self.vmrunpath = Path(virtualboxpath).joinpath("VBoxManage.exe")
+        else:
+            self.vmrunpath = virtualboxpath
+        self.tz = tz
+        self.vminfo = []
+        self.output = []
+
+    def updateOutput(self) -> None:
+        # output = subprocess.run([str(self.vmrunpath), "list", "runningvms"], stdout=subprocess.PIPE)
+        # output = output.stdout.decode("utf-8")
+        with subprocess.Popen([str(self.vmrunpath), "list", "runningvms"], stdout=subprocess.PIPE) as proc:
+            output = proc.stdout.read().decode()
+        if platform.lower() == "win32":
+            output = output.split("\r\n")
+        else:
+            output = output.split("\n")
+        self.output = [{"Name": x[:x.find(' {')].strip().replace('"', ''), "Hash": x[x.find(' {'):].strip()} for x in output if len(x)]
```

Comment:
Does this *need* to be stored? Seems like you can make do with just `vminfo`

Reply:
I ended up building one large list containing dicts of the seperate vm's wit their respective attributes in the form of
```python
output = [vm1:{keys:attributes}, vm2:{keys:attributes}]
```